### DISCUSSION TITLE
What's the difference between the 'before-initialize' and 'initialize' events?

### DISCUSSION THREAD

**Discussion by joelzwarrington** [KonnorRogers/rhino-editor#192] [Q&A] [answered]
I'm wanting to use [Stimulus](https://stimulus.hotwired.dev/) to customize the toolbar via `starterKitOptions`, and the [documentation](https://rhino-editor.vercel.app/how-tos/customizing-the-toolbar/#removing-an-item-from-the-toolbar) suggests using `rhino-before-initialize`, but I believe this event is triggered before Stimulus connects, so I only have access to `rhino-initialize`.

Are there any concerns with modifying the options in there? It seems to work.

**Comment by KonnorRogers** [comment, marked as answer]
@joelzwarrington in theory "rhino-before-initialize" is meant to happen before the editor initializes, I could see a scenario where the hidden input may get overriden too quickly and have a wrong value?

but I've been using "rhino-initialize" and haven't encountered an issue yet so it should be fine, these timing issues are hard. I may even delay the writing to the text input until users have had a full event loop or 2 to be able to attach their extensions before TipTap throws away invalid nodes.

TLDR: should be fine

**Reply by KonnorRogers** [reply]
There's a new way to further "defer initialization" by using an attribute and then removing it. It should help for anyone facing any load order issues.

https://rhino-editor.vercel.app/references/modifying-the-editor/#delaying-initialization
